lib/modules/dosomething/dosomething_api/resources/transactional_resource.inc

```diff
@@ -0,0 +1,76 @@
+<?php
+
+function _transactional_resource_definition() {
+  return [
+    'transactionals' => [
+      'operations' => [
+        'create' => [
+          'help' => 'Create a transactional message.',
+          'file' => [
+            'type' => 'inc',
+            'module' => 'dosomething_api',
+            'name' => 'resources/transactional_resource',
+          ],
+          'callback' => '_transactional_resource_create',
+          'args' => [
+            [
+              'name' => 'id',
+              'type' => 'string',
+              'description' => 'The user\'s Northstar ID.',
+              'source' => 'data',
+              'optional' => FALSE,
+            ],
+            [
+              'name' => 'template',
+              'type' => 'string',
+              'description' => 'The type of transactional to send.',
+              'source' => 'data',
+              'optional' => FALSE,
+            ],
+          ],
+          'access callback' => '_transactional_resource_access',
+          'access arguments' => ['create'],
+        ],
+      ],
+    ]
+  ];
+}
+
+/**
+ * Access callback for User resources.
+ */
+function _transactional_resource_access() {
+  global $user;
+
+  return in_array('administrator', $user->roles);
+}
+
+/**
+ * Callback for triggering a transactional.
+ *
+ * @param array $request
+ *   Array passed to the endpoint. Possible keys:
+ *   - $id (string). The user's Northstar ID
+ *   - $template (string). The type of transactional to send.
+ *
+ * @return mixed
+ */
+function _transactional_resource_create($request) {
+  global $user;
+
+  $account = user_load(dosomething_user_convert_to_legacy_id($request['id']));
```

Comment:
Oh yep, would ya look at that! This is what we've used in the other API endpoints, and is nice since it supports both Northstar IDs and Drupal IDs... it could be nice to combine the two, but I'm always wary of refactoring in Drupal-land.